### DISCUSSION TITLE
Need Help

### DISCUSSION THREAD

**Discussion by it12uw** [OCA/odoo-community.org#115] [General]
![image](https://github.com/OCA/odoo-community.org/assets/135787412/b8dbc413-c61d-4075-8766-8e61dd16158f)

hello guys, do you guys know how to create view like on the picture above?

or you guys have a reference for create a view like that

**Comment by Shide** [comment]
The common way to learn it it's learning how product.attribute.values works with product.templates.

https://github.com/odoo/odoo/blob/16.0/addons/product/views/product_template_views.xml#L59-L75

**Reply by it12uw** [reply]
i have 1 more question
do you know how to disabled a button di kanban box after user click save button in pop up form?

so after user fill all the information and click save, the button in kanban box will be disabled.

kanban box 
![Screenshot 2023-09-30 082119](https://github.com/OCA/odoo-community.org/assets/135787412/823a3e85-0974-4897-8b8f-b8fc70b45137)

pop up form
![Screenshot 2023-09-30 082134](https://github.com/OCA/odoo-community.org/assets/135787412/e77b82af-4e37-443f-9c46-46d0ec3b31ae)

**Reply by theangryangel** [reply]
Probably easiest, but will massively depend on the business logic: add a bool to the record that indicates it’s “locked”. Add that to the kanban and disable/hide the button when true. Bool field could be computed or not. 

Again massively depends on the business logic.

**Reply by it12uw** [reply]
i've try that, but still not working, maybe because the kanban and the form is from 2 diferent model?
or maybe my code is not working properly

here is my code
`is_completed = fields.Boolean(string="Is Completed", default=False)`

`odoo.define('template_qc.layout_shuttle', function (require) {
    "use strict";

    var core = require('web.core');
    var Model = require('web.Model');
    var KanbanView = require('web_kanban.KanbanView');

    var QcKanbanView = KanbanView.extend({
        events: _.extend({}, KanbanView.prototype.events, {
            'click .btn-light': 'onButtonClick',
        }),

        onButtonClick: function (event) {
            var self = this;
            var record_id = $(event.currentTarget).data('id');
            var model = new Model('data.mesin.produksi');

            model.call('toggle_is_completed', [record_id]).then(function () {
                self.reload();
            });
        },
    });

    QcKanbanView.include({
        render: function () {
            this._super.apply(this, arguments);
            var records = this.records;
            _.each(records, function (record) {
                if (record.data.is_completed) {
                    var $button = this.$('.btn-light[data-id="' + record.id + '"]');
                    $button.prop('disabled', true);
                    $button.addClass('disabled');
                }
            }, this);
        },
    });

    core.view_registry.add('layout_shuttle', QcKanbanView);

    return QcKanbanView;
});
`

**Reply by Shide** [reply]
my_app/models/my_model.**py**:
```python
is_completed = fields.Boolean(string="Is Completed", compute="_compute_is_completed", store=True)

@api.depends("one_field", "another_field.one_subfield")
def _compute_is_completed(self):
    for record in self:
        # is_completed = True if all needed fields are filled
        record.is_completed = record.one_field and record.another_field.one_subfield
```

my_app/views/my_model_view.**xml**:
```xml
<kanban>
    <field name="is_completed"/>
    <templates>
        <t t-name="kanban-box">
            <button name="my_func" string="Do something" attrs="{'invisible': [('is_completed', '=', False)]}"/>
        </t>
    </templates>
</kanban>
```
Disabled state cannot be done in xml, so hide button instead.

Try to avoid Javascript unless it's really really necessary. It's hard to maintain.

**Reply by it12uw** [reply]
this code is from mrp module and mrp.production model.

i wondering how this code work.
when i create a new record, the value of this field is 'New', but when i save it the value is changing into WH/MO/00005

the code
`name = fields.Char(
        'Reference', copy=False, readonly=True, default=lambda x: _('New'))`
        
can you explain to me how does that happen?         
because i'm litterally beginer in Odoo ERP.

**Reply by theangryangel** [reply]
These lines:

https://github.com/odoo/odoo/blob/16.0/addons/mrp/models/mrp_production.py#L852-L857

**Reply by it12uw** [reply]
oke thanks a lot

**Comment by Shide** [comment]
I recommend you to do this course first: https://www.odoo.com/slides/technical-training-58

**Reply by it12uw** [reply]
how to display kanban view from model data.mesin.produksi to <notebook> <page> on model snap.qc

kanban view code

`<record id="view_kanban_status" model="ir.ui.view">
        <field name="name">kanban.view.status</field>
        <field name="model">data.mesin.produksi</field>
        <field name="arch" type="xml">
            <kanban default_group_by="line_id"  class="o_kanban_small_column" quick_create="false"  on_create="quick_create" create="0">
                <field name="line_id"/>
                <field name="color"/>
                <field name="is_open_qc_disabled"/>
                <templates>
                    <t t-name="kanban-box">
                        <div t-attf-class="#{kanban_color(record.color.raw_value)} oe_kanban_global_click">
                            <div class="o_dropdown_kanban dropdown">
                                <a role="button" class="dropdown-toggle o-no-caret btn" data-toggle="dropdown" href="#" aria-label="Dropdown menu" title="Klik Disini">
                                    <span class="fa fa-caret-square-o-down"/>
                                </a>
                                <div class="dropdown-menu" role="menu">
                                    <t>
                                        <button class="btn btn-outline-success btn-sm" type="object" name="toggle_disabled_button">
                                            <strong>Checked <i class="fa fa-check-square" aria-hidden="true"></i></strong>
                                        </button>
                                    </t>
                                </div>
                            </div>
                            <div class="oe_kanban_content">
                                <div attrs="{'invisible': [('is_open_qc_disabled', '=', True)]}">
                                    <strong><field name="nomor_mesin"/></strong>  
                                </div>
                                <div class="text-muted">
                                    <strong><field name="kodekain"/></strong>
                                </div>
                            </div>
                        </div>
                    </t>
                </templates>
            </kanban>
        </field>
    </record>` 
    
    i want that kanban view display on this
    
    ` <page string="Layout Shuttle">
                         
                        <p class="oe_grey oe_edit_only">
                            <strong>Warning</strong>: adding or deleting attributes
                            will delete and recreate existing variants and lead
                            to the loss of their possible customizations.
                        </p>`
                        
any idea how to do it?


thankyou

**Reply by it12uw** [reply]
whats the name of this thing in odoo
![image](https://github.com/OCA/odoo-community.org/assets/135787412/1d7dbc28-4a9d-44fa-9a04-36e2771127a0)
